src/plugins/data/public/search/session/session_service.ts

```diff
@@ -203,6 +204,18 @@ export class SessionService {
    * Cleans up current state
    */
   public clear() {
+    // make sure apps can't clear other apps' sessions
+    const currentSessionApp = this.state.get().appName;
+    if (currentSessionApp && currentSessionApp !== this.currentApp) {
```

Comment:
Decided to add this check as a precaution to avoid this possible very bad scenario: 

1. Navigating to another app.
2. Other apps start a search session 
3. First app clears the session 😨